src/mlpack/core/arma_extend/Cube_extra_bones.hpp

```diff
@@ -1,3 +1,3 @@
 //! Add a serialization operator.
-template<typename Archive>
-void serialize(Archive& ar, const unsigned int version);
+template<typename Archive, typename eT>
+void serialize(Archive& ar, arma::Cube<eT>& cube);
```

Comment:
So, actually, we should move all of these `serialize()` functions into a separate `serialization.hpp` file (and possibly `serialization_impl.hpp` if you like), and then include that directly from `arma_extend.hpp`.  Everything in, e.g., `Cube_extra_bones.hpp` will be included *directly in* the definition of the `arma::Cube` class.  Ideally, like @conradsnicta suggested, we should remove these entirely.  We'll be most of the way there after this PR and then I can clean up the last bits in a different PR.

Let me know if anything I wrote doesn't make sense there. :+1: